### PR TITLE
Handle the case that a storage accidentally lost

### PR DESCRIPTION
**Pull request by ls-zhu**
bsc#1085421. Handle the case that the storage of a LUN
accidentally get disconnected.

### CODE REVIEW

File: src/include/iscsi-lio-server/TargetData.rb [hunk 1]
```diff
@@ -462,7 +466,11 @@ def analyze
         lun_num_int = lun_num[3,lun_num.length]
         lun_path_tmp = RE_LUN_PATH.match(line).to_s
         lun_path = lun_path_tmp[1,lun_path_tmp.length-2]
-        @current_tpg.store_lun(lun_num,[rand(9999), lun_num_int, lun_name, lun_path, File.ftype(lun_path)])
+        if !File.exist?(lun_path)
+          Yast::Popup.Error(_("Can not access the storage ") + lun_path + _(". Please consider re-connect the storage or delete then  re-create the target which using this storage.\n"))
```
Comment (mvidner) [inline src/include/iscsi-lio-server/TargetData.rb]: Use this:

```rb
msg = format(_("Cannot access the storage %s.\n" \
  "Please consider reconnecting the storage or\n" \
  "deleting then recreating the target which is using this storage."), lun_path)
Yast::Popup.Error(msg)
```

https://ruby-doc.org/core-2.1.0/Kernel.html#method-i-format

**Comment by mvidner** [comment]
Explanation of the localization problem:

Adding translated strings does not work, because other languages have different word order.
For example, "Could not connect to host suse.com" comes out in Japanese grammar roughly  as "suse.com-of host-with connect could not", so `_("Could not connect to host ") + host` cannot be translated satisfactorily. You have to use `format(_("Could not connect to host %s"), host)` which the translator can translate to "%s-of host-with connect could not".

**Comment by ls-zhu** [comment]
Thanks Martin, new commit pushed.

**Comment by ls-zhu** [comment]
Thanks Martin!

**Pull request merged**